com.microsoft.mrtk.tools/SubsystemWizard/SubsystemGenerator.cs

```diff
@@ -0,0 +1,605 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.CSharp;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Tools
+{
+    /// <summary>
+    /// Class which generates subsystem source files from templates contained within
+    /// the package.
+    /// </summary>
+    [Serializable]
+    internal class SubsystemGenerator
+    {
+        // SubsystemWizard/Templates/ApplyConfigTemplate.txt
+        private const string ApplyConfigTemplateGuid = "c4bce3779fa330840bc15f96d11fcec0";
+        // SubsystemWizard/Templates/SubsystemBaseClassTemplate.txt
+        private const string BaseClassTemplateGuid = "2cb3a153fb7b8b142a1dac1bcfc3b38c";
+        // SubsystemWizard/Templates/SubsystemConfigTemplate.txt
+        private const string ConfigTemplateGuid = "ab5924fb380e64d47bdbd9fed4c08910";
+        // SubsystemWizard/Templates/SubsystemDerivedClassTemplate.txt
+        private const string DerivedClassTemplateGuid = "02b67503c64cf224ca2c04a04077c0d7";
+        // SubsystemWizard/Templates/SubsystemDescriptorTemplate.txt
+        private const string DescriptorTemplateGuid = "8e1afb29fbaf1c2419511d266f49b976";
+        // SubsystemWizard/Templates/SubsystemInterfaceTemplate.txt
+        private const string InterfaceTemplateGuid = "9bac30e514266984d947e360cfd17b05";
+
+        private const bool DefaultCreateConfiguration = false;
+        private static readonly string DefaultBaseSubsystemName = $"NewSubsystem";
+        private static readonly string DefaultOrganizationName = "Contoso";
```

Comment:
Would we maybe want to default to https://docs.unity3d.com/ScriptReference/PlayerSettings-companyName.html?

Reply:
ooh. i like that.

Reply:
done. thx for the idea!